packages/fastboot/src/sandbox.js

```diff
@@ -53,6 +58,85 @@ module.exports = class Sandbox {
     return wrappedConsole;
   }
 
+  buildFetch() {
+    let globals;
+
+    if (globalThis.fetch) {
+      globals = {
+        fetch: globalThis.fetch,
+        Request: globalThis.Request,
+        Response: globalThis.Response,
+        Headers: globalThis.Headers,
+        AbortController: globalThis.AbortController,
+      };
+    } else {
+      let nodeFetch = require('node-fetch');
+      let {
+        AbortController,
+        abortableFetch,
+      } = require('abortcontroller-polyfill/dist/cjs-ponyfill');
+      let { fetch, Request } = abortableFetch({
+        fetch: nodeFetch,
+        Request: nodeFetch.Request,
+      });
+
+      globals = {
+        fetch,
+        Request,
+        Response: nodeFetch.Response,
+        Headers: nodeFetch.Headers,
+        AbortController,
+      };
+    }
+
+    let originalFetch = globals.fetch;
+    globals.fetch = function __fastbootFetch(input, init) {
+      if (input && input.href) {
+        input.url = globals.fetch.__fastbootBuildAbsoluteURL(input.href);
+      } else if (typeof input === 'string') {
+        input = globals.fetch.__fastbootBuildAbsoluteURL(input);
+      }
+      return originalFetch(input, init);
+    };
+
+    globals.fetch.__fastbootBuildAbsoluteURL = function __fastbootBuildAbsoluteURL(url) {
+      if (protocolRelativeRegex.test(url)) {
+        let [host] = globals.fetch.__fastbootParseRequest(url, fetch.__fastbootRequest);
+        url = `${host}${url}`;
+      } else if (!httpRegex.test(url)) {
+        let [host, protocol] = globals.fetch.__fastbootParseRequest(url, fetch.__fastbootRequest);
+        url = `${protocol}//${host}${url}`;
+      }
+      return url;
+    };
+
+    globals.fetch.__fastbootParseRequest = function __fastbootParseRequest(url, request) {
+      if (!request) {
+        throw new Error(
+          `Using fetch with relative URL ${url}, but application instance has not been initialized yet.`
+        );
+      }
+      // Old Prember version is not sending protocol
```

Comment:
@gabrielgrant, should be old versions of prember still supported? Based on the changelog the 1.0.2+ send protocol correctly https://github.com/ef4/prember/blob/master/CHANGELOG.md#102---2019-01-06

Reply:
I see. Is that patch incompatible with the changes you are making?

Reply:
Nope, this workaround should still work for `prember < 1.0.2`, hence it can stay here if it make sense 🙌

Reply:
I would guess most prember users are on the latest version but if the workaround can stay that is ideal